widgetssdk/src/main/java/com/glia/widgets/call/CallView.java

```diff
@@ -644,18 +652,26 @@ private void showScreenSharingDialog() {
     }
 
     private void setButtonActivated(FloatingActionButton floatingActionButton,
-                                    Integer activatedDrawableRes,
-                                    Integer notActivatedDrawableRes,
+                                    TextView textView,
                                     @StringRes int activatedContentDescription,
                                     @StringRes int notActivatedContentDescription,
+                                    boolean isEnabled,
                                     boolean isActivated
     ) {
         floatingActionButton.setActivated(isActivated);
-        floatingActionButton.setImageResource(isActivated ? activatedDrawableRes : notActivatedDrawableRes);
         floatingActionButton.setContentDescription(resources.getString(isActivated ?
                 activatedContentDescription :
                 notActivatedContentDescription
         ));
+        floatingActionButton.setEnabled(isEnabled);
+        setBarButtonLabelAppearance(
+                textView,
+                isActivated ?
+                        StyleUtils.FabButtonStatus.SELECTED :
+                        isEnabled ?
+                                StyleUtils.FabButtonStatus.ACTIVE :
+                                StyleUtils.FabButtonStatus.INACTIVE
```

Comment:
I don't think this is reasonable or readable to pass as a second argument a double-inline-if statement )
Can you please try to extract it somehow?